### ISSUE TITLE
SMC Battery manger hanging

### ISSUE THREAD

**Issue by Edwardwich** [1Revenger1/ECEnabler#24]
I have problem with VirtualSMC battery kext :
I use SMC Battery kext 
I add Ecenable kext but nothing change 

sometime when laptop connect to ac :
![133926993-fdcae46c-bec1-414c-805d-20f9d73f51a2](https://user-images.githubusercontent.com/35195176/208325443-291c6b5a-0c92-4d29-a8e6-9f19122e0f1e.png)

and charging light when back to red (charging battery) :
![image](https://user-images.githubusercontent.com/35195176/208325499-dfeb790a-5fb0-485a-9d15-6260228e919d.png)
battery person Percentage not change only way back to work unplug and plug agin

ioreg :
[ali’s MacBook Pro.zip](https://github.com/1Revenger1/ECEnabler/files/10254952/ali.s.MacBook.Pro.zip)
apci:
https://github.com/Edwardwich/hp-DA1023nia-macOS/tree/main/Opencore-Debug/SysReport/ACPI
config:
[config.plist.zip](https://github.com/1Revenger1/ECEnabler/files/10254946/config.plist.zip)

**Comment by 1Revenger1** [comment]
No, I've got no clue why this would happen. All ECEnabler does is read big ACPI fields.

**Comment by Edwardwich** [comment]
> No, I've got no clue why this would happen. All ECEnabler does is read big EC fields.

ok TNX

**Comment by MinecraftNerrd1337** [comment]
What device is it?

**Comment by Edwardwich** [comment]
@1Revenger1 HP DA1023nia its a main laptop I use
https://github.com/Edwardwich/hp-DA1023nia-macOS

**Comment by MinecraftNerrd1337** [comment]
As @1Revenger1 said, all ECEnabler does is read big EC fields. Some devices will still require further ACPI patching, which can be a mess because it's completely different for every device.

I will need to do this for the Lenovo Legion I'm currently writing from (15ACH6H) because battery status freezes after wake from sleep. So I'll learn sh*t about this topic and possibly end up being able to help people with it

**Comment by Edwardwich** [comment]
> As @1Revenger1 said, all ECEnabler does is read big EC fields. Some devices will still require further ACPI patching, which can be a mess because it's completely different for every device.
> 
> I will need to do this for the Lenovo Legion I'm currently writing from (15ACH6H) because battery status freezes after wake from sleep. So I'll learn sh*t about this topic and possibly end up being able to help people with it

ok I need ur help all ssdt and dsdt of my laptop I post it on my GitHub repo :
https://github.com/Edwardwich/hp-DA1023nia-macOS/tree/main/Opencore-Debug/SysReport
